nni/algorithms/compression/pytorch/auto_compress/utils.py

```diff
@@ -0,0 +1,126 @@
+        for config in config_list:
+            op_types = config.pop('op_types', [])
+            op_names = config.pop('op_names', [])
+            key_prefix = 'config_list::{}::{}'.format(':'.join(op_types), ':'.join(op_names))
+            for var_name, var_search_space in config.items():
+                sub_search_space['{}::{}'.format(key_prefix, var_name)] = self._wrap_single_value(var_search_space)
+        for parameter_name, parameter_search_space in algo_kwargs.items():
+            key_prefix = 'parameter'
+            sub_search_space['{}::{}'.format(key_prefix, parameter_name)] = self._wrap_single_value(parameter_search_space)
+        self.pruner_choice_list.append(sub_search_space)
+
+    def add_quantizer_config(self, quantizer_name: str, config_list: list, **algo_kwargs):
+        """
+        Parameters
+        ----------
+        quantizer_name
+            Supported pruner name: 'naive', 'qat', 'dorefa', 'bnn'.
+        config_list
+            Except 'quant_types', 'op_types' and 'op_names', other config value can be written as `{'_type': ..., '_value': ...}`.
+        **algo_kwargs
+            The additional pruner parameters except 'model', 'config_list', 'optimizer'.
+        """
+        sub_search_space = {'_name': quantizer_name}
+        for config in config_list:
+            quant_types = config.pop('quant_types', [])
+            op_types = config.pop('op_types', [])
+            op_names = config.pop('op_names', [])
+            key_prefix = 'config_list::{}::{}::{}'.format(':'.join(quant_types), ':'.join(op_types), ':'.join(op_names))
+            for var_name, var_search_space in config.items():
+                sub_search_space['{}::{}'.format(key_prefix, var_name)] = self._wrap_single_value(var_search_space)
+        for parameter_name, parameter_search_space in algo_kwargs.items():
+            key_prefix = 'parameter'
+            sub_search_space['{}::{}'.format(key_prefix, parameter_name)] = self._wrap_single_value(parameter_search_space)
+        self.quantizer_choice_list.append(sub_search_space)
+
+    def dumps(self) -> dict:
+        """
+        Dump the search space as a dict.
+        """
+        compressor_choice_value = []
+        if self.pruner_choice_list:
+            compressor_choice_value.append({
+                '_name': 'pruner',
+                'algorithm_name': {
+                    '_type': 'choice',
+                    '_value': self.pruner_choice_list
+                }
+            })
+        if self.quantizer_choice_list:
+            compressor_choice_value.append({
+                '_name': 'quantizer',
+                'algorithm_name': {
+                    '_type': 'choice',
+                    '_value': self.quantizer_choice_list
+                }
+            })
+        search_space = {
+            'compressor_type': {
+                '_type': 'choice',
+                '_value': compressor_choice_value
+            }
+        }
```

Comment:
it is better to merge pruner_choice_list and quantizer_choice_list together, then the nest depth is reduced by one, which is simpler and is friendly to tuning algorithms

Reply:
merge these.